react/components/Popup.tsx

```diff
@@ -16,26 +16,41 @@ const CSS_HANDLES = [
 ]
 
 const PopupMode: FC = ({ children }) => {
-  const { open, hasBeenOpened, openOnHover } = useMinicartState()
+  const {
+    open,
+    hasBeenOpened,
+    openOnHoverBehavior,
+    openOnHoverProp,
+  } = useMinicartState()
   const dispatch = useMinicartDispatch()
   const handles = useCssHandles(CSS_HANDLES)
 
+  const handleClick = () => {
+    if (openOnHoverProp) {
+      dispatch({ type: 'SET_OPEN_ON_HOVER_BEHAVIOR', value: true })
+    }
+    dispatch({ type: 'CLOSE_MINICART' })
+  }
+  const handleMouseLeave = () => {
+    if (!openOnHoverBehavior) {
+      return
+    }
+    dispatch({ type: 'CLOSE_MINICART' })
+  }
+
   return (
-    <Fragment>
+    <div onMouseLeave={handleMouseLeave}>
```

Comment:
Can you add this callback just if needed? Like you did in the component?

Reply:
Sure!